File: lib/parser/packet/ProfileParameter.java
package com.yuhki50.lpr9201.parser.packet;

import com.yuhki50.lpr9201.util.ByteUtil;

/**
 * プロファイルパラメータ確認
 */
public class ProfileParameter implements IParsePacket {
    /**
     * 結果コード
     */
    protected static final int RESULT_CODE = 0x85;

    /**
     * データ長のバイト数
     */
    protected static final int DATA_LENGTH_BYTE_SIZE = 1;

    /**
     * 受信したリザルトデータ
     */
    protected Result result;

    /**
     * プロファイルパラメータ確認
     *
     * @param result 結果クラス
     */
    public ProfileParameter(Result result) {
        this.result = result;
    }

    /**
     * 結果コードを取得する
     *
     * @return 結果コード
     */
    public int getResultCode() {
        return RESULT_CODE;
    }

    /**
     * データ長のバイト数を取得する
     *
     * @return データ長のバイト数
     */
    public int getDataLengthByteSize() {
        return DATA_LENGTH_BYTE_SIZE;
    }

    /**
     * パース可能か
     *
     * @return true:パース可能, false: パース不可
     */
    public boolean canParse() {
        return this.result != null && this.result.resultCode == RESULT_CODE;
    }

    /**
     * プロファイルパラメーター値を取得する
     *
     * @return プロファイルパラメーター値
     */
    public int getValueInt() {
        return (int) ByteUtil.mergeBigEndian(this.result.datas);
    }

    /**
     * プロファイルパラメーター値を取得する
     *
     * @return プロファイルパラメーター値
     */
    public long getValueLong() {
        return ByteUtil.mergeBigEndian(this.result.datas);
    }

    /**
     * プロファイルパラメーター値を取得する
     *
     * @return プロファイルパラメーター値
     */
    public int[] getValueBytes() {
        return this.result.datas;
    }
}

File: src/parser/packet/ProfileParameter.ts
/// <reference path='IParsePacket.ts' />
/// <reference path='Result.ts' />
/// <reference path='../../util/ByteUtil.ts' />

/**
 * プロファイルパラメータ確認
 */
class ProfileParameter implements IParsePacket {
    /**
     * 結果コード
     */
    //protected static const RESULT_CODE : number = 0x85;
    protected static RESULT_CODE : number = 0x85;

    /**
     * データ長のバイト数
     */
    //protected static const DATA_LENGTH_BYTE_SIZE : number = 1;
    protected static DATA_LENGTH_BYTE_SIZE : number = 1;

    /**
     * 受信したリザルトデータ
     */
    protected result : Result;

    /**
     * プロファイルパラメータ確認
     *
     * @param result 結果クラス
     */
    public constructor(result : Result) {
        this.result = result;
    }

    /**
     * 結果コードを取得する
     *
     * @return 結果コード
     */
    public getResultCode() : number {
        return ProfileParameter.RESULT_CODE;
    }

    /**
     * データ長のバイト数を取得する
     *
     * @return データ長のバイト数
     */
    public getDataLengthByteSize() : number {
        return ProfileParameter.DATA_LENGTH_BYTE_SIZE;
    }

    /**
     * パース可能か
     *
     * @return true:パース可能, false: パース不可
     */
    public canParse() : boolean {
        return this.result != null && this.result.resultCode == ProfileParameter.RESULT_CODE;
    }

    /**
     * プロファイルパラメーター値を取得する
     *
     * @return プロファイルパラメーター値
     */
    public getValueInt() : number {
        return ByteUtil.mergeBigEndian(this.result.datas);
    }

    /**
     * プロファイルパラメーター値を取得する
     *
     * @return プロファイルパラメーター値
     */
    public getValueLong() : number {
        return ByteUtil.mergeBigEndian(this.result.datas);
    }

    /**
     * プロファイルパラメーター値を取得する
     *
     * @return プロファイルパラメーター値
     */
    public getValueBytes() : number[] {
        return this.result.datas;
    }
}
